includes/discovery/sensors/temperature/comware.inc.php

```diff
@@ -8,11 +8,16 @@
  * Free Software Foundation, either version 3 of the License, or (at your
  * option) any later version.  Please see LICENSE.txt at the top level of
  * the source code distribution for details.
- */
-
+ *
+ * @author     Peca Nesovanovic <peca.nesovanovic@sattrakt.com>
+*/
 echo 'Comware ';
 
-$entphydata = dbFetchRows("SELECT `entPhysicalIndex`, `entPhysicalClass`, `entPhysicalName` FROM `entPhysical` WHERE `device_id` = ? AND `entPhysicalClass` REGEXP 'module|sensor' ORDER BY `entPhysicalIndex`", [$device['device_id']]);
+$entphydata = DeviceCache::getPrimary()->entityPhysical()
+    ->where('entPhysicalClass', 'regexp', 'module|sensor')
```

Comment:
This can probably have a tiny performance win by
```suggestion
    ->whereIn('entPhysicalClass', ['module', 'sensor'])
```
but untested.

Reply:
tried that at first run, before commit, but then changed to regexp, to be same as original :)
yes, it is working boh way